mod_ci/controllers.py

```diff
@@ -555,43 +555,48 @@ def start_ci():
 
         elif event == "pull_request":
             # If it's a PR, run the tests
-            commit = ''
-            gh_commit = None
-            if payload['action'] in ['opened', 'synchronize']:
-                try:
-                    commit = payload['pull_request']['head']['sha']
-                    gh_commit = repository.statuses(commit)
-                except KeyError:
-                    g.log.critical("Didn't find a SHA value for a newly opened PR!")
-                    g.log.debug(payload)
-            elif payload['action'] == 'closed':
-                g.log.debug('PR was closed, no after hash available')
-
-            pr_nr = payload['pull_request']['number']
-            if payload['action'] == 'opened':
-                # Run initial tests
-                queue_test(g.db, gh_commit, commit, TestType.pull_request, pr_nr=pr_nr)
-            elif payload['action'] == 'synchronize':
-                # Run/queue a new test set
-                queue_test(g.db, gh_commit, commit, TestType.pull_request, pr_nr=pr_nr)
-            elif payload['action'] == 'closed':
-                # Cancel running queue
-                tests = Test.query.filter(Test.pr_nr == pr_nr).all()
-                for test in tests:
-                    # Add canceled status only if the test hasn't started yet
-                    if len(test.progress) > 0:
-                        continue
-                    progress = TestProgress(test.id, TestStatus.canceled, "PR closed", datetime.datetime.now())
-                    g.db.add(progress)
-                    repository.statuses(test.commit).post(
-                        state=Status.FAILURE,
-                        description="Tests canceled",
-                        context="CI - {name}".format(name=test.platform.value),
-                        target_url=url_for('test.by_id', test_id=test.id, _external=True)
-                    )
-            elif payload['action'] == 'reopened':
-                # Run tests again
-                queue_test(g.db, gh_commit, commit, TestType.pull_request, pr_nr=pr_nr)
+            # Check if user blacklisted
+            if BlockedUsers.query.filter(BlockedUsers.userID == payload[
+                   'pull_request']['user']['id']).first():
```

Comment:
No trace besides a single log line, and user doesn't get notified? Please change.

Reply:
Added return.